crates/gitbutler-settings/vendor/util/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+use std::borrow::Cow;
+
+use rust_embed::Embed;
+
+pub fn merge_non_null_json_value_into(source: serde_json::Value, target: &mut serde_json::Value) {
```

Comment:
The overhead to get this amount of code seems disproportional.

My suggestion is to adapt it, and make it better. The problem is that it can't be much different for what it's doing, but I tried anyway, and added some tests:

```patch
commit 5a8c8505f768c06eeaae3cfd419fa4dfb198e51c
Author: Sebastian Thiel <sebastian.thiel@icloud.com>
Date:   Thu Dec 12 08:13:39 2024 +0100

    tmp

diff --git a/Cargo.lock b/Cargo.lock
index f3527b413..652855d09 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2833,7 +2833,6 @@ version = "0.0.0"
 dependencies = [
  "anyhow",
  "gitbutler-fs",
- "gitbutler-settings",
  "gitbutler-settings-util",
  "rust-embed",
  "serde",
diff --git a/crates/gitbutler-settings/Cargo.toml b/crates/gitbutler-settings/Cargo.toml
index 8dfa14118..7257338bd 100644
--- a/crates/gitbutler-settings/Cargo.toml
+++ b/crates/gitbutler-settings/Cargo.toml
@@ -18,5 +18,3 @@ gitbutler-fs.workspace = true
 name = "settings"
 path = "tests/mod.rs"
 
-[dev-dependencies]
-gitbutler-settings.workspace = true
diff --git a/crates/gitbutler-settings/src/persistence.rs b/crates/gitbutler-settings/src/persistence.rs
index 957580480..ac6578a71 100644
--- a/crates/gitbutler-settings/src/persistence.rs
+++ b/crates/gitbutler-settings/src/persistence.rs
@@ -3,7 +3,7 @@ use std::path::PathBuf;
 use crate::app_settings::AppSettings;
 use crate::json::json_difference;
 use anyhow::Result;
-use gitbutler_settings_util::{asset_str, merge_non_null_json_value_into};
+use gitbutler_settings_util::{asset_str, merge_non_null_json_value};
 use rust_embed::Embed;
 use serde_json::json;
 
@@ -27,7 +27,7 @@ impl AppSettings {
         let customizations = serde_json_lenient::from_str(&std::fs::read_to_string(config_path)?)?;
 
         // Merge the customizations into the settings
-        merge_non_null_json_value_into(customizations, &mut settings);
+        merge_non_null_json_value(customizations, &mut settings);
         Ok(serde_json::from_value(settings)?)
     }
 
@@ -53,7 +53,7 @@ impl AppSettings {
 
         // Merge the new customizations into the existing ones
         // TODO: This will nuke any comments in the file
-        merge_non_null_json_value_into(diff, &mut customizations);
+        merge_non_null_json_value(diff, &mut customizations);
         gitbutler_fs::write(config_path, customizations.to_string())?;
         Ok(())
     }
diff --git a/crates/gitbutler-settings/vendor/util/src/lib.rs b/crates/gitbutler-settings/vendor/util/src/lib.rs
index d4e3d02d5..30d2204f4 100644
--- a/crates/gitbutler-settings/vendor/util/src/lib.rs
+++ b/crates/gitbutler-settings/vendor/util/src/lib.rs
@@ -2,20 +2,22 @@ use std::borrow::Cow;
 
 use rust_embed::Embed;
 
-pub fn merge_non_null_json_value_into(source: serde_json::Value, target: &mut serde_json::Value) {
+/// Based on Zed `merge_non_null_json_value_into`
+/// Note: This doesn't merge arrays.
+pub fn merge_non_null_json_value(source: serde_json::Value, target: &mut serde_json::Value) {
     use serde_json::Value;
     if let Value::Object(source_object) = source {
         let target_object = if let Value::Object(target) = target {
             target
         } else {
-            *target = Value::Object(Default::default());
-            target.as_object_mut().unwrap()
+            *target = serde_json::json!({});
+            target.as_object_mut().expect("object was just set")
         };
         for (key, value) in source_object {
             if let Some(target) = target_object.get_mut(&key) {
-                merge_non_null_json_value_into(value, target);
+                merge_non_null_json_value(value, target);
             } else if !value.is_null() {
-                target_object.insert(key.clone(), value);
+                target_object.insert(key, value);
             }
         }
     } else if !source.is_null() {
@@ -23,6 +25,36 @@ pub fn merge_non_null_json_value_into(source: serde_json::Value, target: &mut se
     }
 }
 
+// TODO: put into ./tests/
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn it_does_not_merge_null_values() {
+        let source = serde_json::json!({"a": null, "b": true });
+        let mut target = serde_json::json!({});
+        merge_non_null_json_value(source, &mut target);
+        assert_eq!(target, serde_json::json!({"b": true }));
+    }
+
+    #[test]
+    fn it_does_not_merge_arrays() {
+        let source = serde_json::json!({"a": null, "b": [1,2,3]});
+        let mut target = serde_json::json!({"a": {"b": 1}, "b": [42]});
+        merge_non_null_json_value(source, &mut target);
+        assert_eq!(target, serde_json::json!({"a": {"b": 1}, "b": [1,2,3] }));
+    }
+
+    #[test]
+    fn it_merges_nested_objects_correctly() {
+        let source = serde_json::json!({"a": {"b": {"c": 42}}});
+        let mut target = serde_json::json!({});
+        merge_non_null_json_value(source.clone(), &mut target);
+        assert_eq!(target, source);
+    }
+}
+
 /// Get an embedded file as a string.
 pub fn asset_str<A: Embed>(path: &str) -> Cow<'static, str> {
     match A::get(path).unwrap().data {
```

That way the whole sub-crate could go away, which I'd appreciate.